spec/support/requests_shared_examples.rb

```diff
@@ -213,6 +213,32 @@
   end
 end
 
+shared_examples 'a sorted index resource' do |expected_last_item_sym|
+  let(:expected_last_item) { send(expected_last_item_sym) }
+  let(:sort_column) { :created_at }
+  let(:sort_order) { "desc" }
+
+  it 'should return a sorted index' do
+    is_expected.to eq(expected_response_status)
+    response_json = JSON.parse(response.body)
+    expect(response_json).to have_key('results')
+    returned_results = response_json['results']
+    expect(returned_results).to be_a(Array)
+    last_date = nil
+    returned_results.each do |this_result|
+      this_result_object = KindnessFactory.by_kind(this_result["kind"]).find(this_result["id"])
+      if last_date.nil?
+        last_date = this_result_object.send(sort_column)
+      elsif sort_order == "asc"
+        expect(this_result_object.send(sort_column)).to be >= last_date
+      else
+        expect(this_result_object.send(sort_column)).to be <= last_date
+      end
+    end
+    expect(returned_results.last["id"]).to eq(expected_last_item.id)
```

Comment:
I don't think that you need this `expected_last_item` test. If you want to keep it though, it would be better suited in a separate `it` block.

Reply:
I'll put it in its own it block. I like it because it forces the test writer to do something specific to an item to ensure it is not part of a `let (:items) { FactoryBot.create_list(:foo, 3) }` group which might have very similar created_at/updated_at entries.